on_startup.py

```diff
@@ -42,69 +74,29 @@ def populate_higlass_data_directory(data_dir):
 
 def ingest_tilesets(data_dir):
     """
-   Ingest previously downloaded files into higlass-server w/ django
-   management command
-   :param data_dir: <String> Path to directory populated with data to ingest
-   """
-    files_to_ingest = glob.glob(
-        '{}*multires.*'.format(data_dir)
-    )
-
-    for filename in files_to_ingest:
+    Ingest previously downloaded files into higlass-server w/ django
+    management command
+    :param data_dir: <String> Path to directory populated with data to ingest
+    """
+    for refinery_node_uuid in TILE_SETS.keys():
+        refinery_node_info = TILE_SETS[refinery_node_uuid]
+        file_path = data_dir + refinery_node_info["file_name"]
+
         call_command(
             "ingest_tileset",
-            filename="{}".format(filename),
-            filetype=get_filetype(filename),
-            datatype=get_datatype(filename)
+            filename=file_path,
+            filetype=refinery_node_info[HIGLASS_FILE_TYPE],
+            datatype=refinery_node_info[HIGLASS_DATA_TYPE]
         )
 
 
-def get_datatype(filename):
-    datatype_mapping = {
-        "cool": "matrix",
-        "hitile": "vector"
-    }
-    try:
-        datatype = datatype_mapping[filename.split(".")[-1]]
-    except KeyError as e:
-        raise RuntimeError(
-            "Could not determine datatype from filename: {}".format(
-                filename
-            )
-        )
-    else:
-        return datatype
-
-
-def get_filetype(filename):
-    filetype_mapping = {
-        "cool": "cooler",
-        "hitile": "hitile"
-    }
-    try:
-        filetype = filetype_mapping[filename.split(".")[-1]]
-    except KeyError:
-        raise RuntimeError(
-            "Could not determine filetype from filename: {}".format(
-                filename
-            )
-        )
-    else:
-        return filetype
-
-
-def swap_waiting_page():
-    subprocess.call(["mv", "/home/higlass/projects/higlass-website/app/index.html",
-                     "/home/higlass/projects/higlass-website/index.html"])
-
 if __name__ == '__main__':
     data_dir = "/refinery-data/"
-
     # Allows for django commands to run in a standalone script
     django.setup()
 
     populate_higlass_data_directory(data_dir)
     ingest_tilesets(data_dir)
 
-    # Don't switch page until data ingested
-    swap_waiting_page()
+    # Start Nginx only after tilesets have been ingested
+    subprocess.run(["/usr/sbin/nginx"])
```

Comment:
You sort of need to jump through hoops to completely hand off to a new process, but perhaps consider: https://stackoverflow.com/questions/3516007/run-process-and-dont-wait

(In perl `exec` did this, so I was wondering if there were an analog in python.)

Reply:
Comment that parent will hang around